File: app/src/main/java/org/androidforfun/droids/view/Assets.java
/*
 *  Copyright (C) 2016 Salvatore D'Angelo
 *  This file is part of Droids project.
 *  This file derives from the Mr Nom project developed by Mario Zechner for the Beginning Android
 *  Games book (chapter 6).
 *
 *  Droids is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  Droids is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License.
 */
package org.androidforfun.droids.view;

import org.androidforfun.framework.Music;
import org.androidforfun.framework.Pixmap;
import org.androidforfun.framework.Sound;

/*
 * This class contains the global references to all the assets used in DroidsWorld game.
 *
 * @author Salvatore D'Angelo
 */
public class Assets {
    // the logo asset
    public static Pixmap logo;

    // the screen used in DroidsWorld game
    public static Pixmap gamescreen;
    public static Pixmap startscreen;
    public static Pixmap highscoresscreen;
    public static Pixmap gameoverscreen;

    // the menu used in DroidsWorld game
    public static Pixmap mainmenu;
    public static Pixmap pausemenu;
    public static Pixmap readymenu;

    // these are the colored block to draw the DroidsWorld shape. Each shape is composed by 4 blocks
    // of same colors. Each shape has a different color.
    public static Pixmap redblock;
    public static Pixmap greenblock;
    public static Pixmap blueblock;
    public static Pixmap cyanblock;
    public static Pixmap yellowblock;
    public static Pixmap magentablock;
    public static Pixmap orangeblock;

    // these are the colored block to draw the DroidsWorld next shape.
    public static Pixmap smallredblock;
    public static Pixmap smallgreenblock;
    public static Pixmap smallblueblock;
    public static Pixmap smallcyanblock;
    public static Pixmap smallyellowblock;
    public static Pixmap smallmagentablock;
    public static Pixmap smallorangeblock;

    // buttons and numbers
    public static Pixmap buttons;
    public static Pixmap numbers;

    // sounds
    public static Sound click;
    public static Sound bitten;

    // music
    public static Music music;

    public static Pixmap getBlockByColor(int color) {
        switch(color) {
            case 0xffffff00: return Assets.yellowblock;
            case 0xffb2ffff: return Assets.cyanblock;
            case 0xff0000ff: return Assets.blueblock;
            case 0xffff7f00: return Assets.orangeblock;
            case 0xff00ff00: return Assets.greenblock;
            case 0xffff00ff: return Assets.magentablock;
            case 0xffff0000: return Assets.redblock;
            default: return Assets.redblock;
        }
    }

    public static Pixmap getSmallBlockByColor(int color) {
        switch(color) {
            case 0xffffff00: return Assets.smallyellowblock;
            case 0xffb2ffff: return Assets.smallcyanblock;
            case 0xff0000ff: return Assets.smallblueblock;
            case 0xffff7f00: return Assets.smallorangeblock;
            case 0xff00ff00: return Assets.smallgreenblock;
            case 0xffff00ff: return Assets.smallmagentablock;
            case 0xffff0000: return Assets.smallredblock;
            default: return Assets.smallredblock;
        }
    }
}

File: app/src/main/kotlin/org/code4projects/droids/view/Assets.kt
/*
 * Copyright (c) 2016-2026 Salvatore D'Angelo
 * Licensed under the MIT License. See the LICENSE file in the project root.
 */
package org.code4projects.droids.view

import org.code4projects.framework.Music
import org.code4projects.framework.Pixmap
import org.code4projects.framework.Sound

/*
 * This class contains the global references to all the assets used in DroidsWorld game.
 *
 * @author Salvatore D'Angelo
 */
object Assets {
    // the logo asset
    @JvmField var logo: Pixmap? = null

    // the screen used in DroidsWorld game
    @JvmField var gamescreen: Pixmap? = null
    @JvmField var startscreen: Pixmap? = null
    @JvmField var highscoresscreen: Pixmap? = null
    @JvmField var gameoverscreen: Pixmap? = null

    // the menu used in DroidsWorld game
    @JvmField var mainmenu: Pixmap? = null
    @JvmField var pausemenu: Pixmap? = null
    @JvmField var readymenu: Pixmap? = null

    // these are the colored block to draw the DroidsWorld shape. Each shape is composed by 4 blocks
    // of same colors. Each shape has a different color.
    @JvmField var redblock: Pixmap? = null
    @JvmField var greenblock: Pixmap? = null
    @JvmField var blueblock: Pixmap? = null
    @JvmField var cyanblock: Pixmap? = null
    @JvmField var yellowblock: Pixmap? = null
    @JvmField var magentablock: Pixmap? = null
    @JvmField var orangeblock: Pixmap? = null

    // these are the colored block to draw the DroidsWorld next shape.
    @JvmField var smallredblock: Pixmap? = null
    @JvmField var smallgreenblock: Pixmap? = null
    @JvmField var smallblueblock: Pixmap? = null
    @JvmField var smallcyanblock: Pixmap? = null
    @JvmField var smallyellowblock: Pixmap? = null
    @JvmField var smallmagentablock: Pixmap? = null
    @JvmField var smallorangeblock: Pixmap? = null

    // buttons and numbers
    @JvmField var buttons: Pixmap? = null
    @JvmField var numbers: Pixmap? = null

    // sounds
    @JvmField var click: Sound? = null
    @JvmField var bitten: Sound? = null

    // music
    @JvmField var music: Music? = null

    @JvmStatic
    fun getBlockByColor(color: Int): Pixmap? = when (color) {
        0xffffff00L.toInt() -> yellowblock
        0xffb2ffffL.toInt() -> cyanblock
        0xff0000ffL.toInt() -> blueblock
        0xffff7f00L.toInt() -> orangeblock
        0xff00ff00L.toInt() -> greenblock
        0xffff00ffL.toInt() -> magentablock
        0xffff0000L.toInt() -> redblock
        else -> redblock
    }

    @JvmStatic
    fun getSmallBlockByColor(color: Int): Pixmap? = when (color) {
        0xffffff00L.toInt() -> smallyellowblock
        0xffb2ffffL.toInt() -> smallcyanblock
        0xff0000ffL.toInt() -> smallblueblock
        0xffff7f00L.toInt() -> smallorangeblock
        0xff00ff00L.toInt() -> smallgreenblock
        0xffff00ffL.toInt() -> smallmagentablock
        0xffff0000L.toInt() -> smallredblock
        else -> smallredblock
    }
}
